### PR TITLE
More helpful error message for invalid formats

### PR DESCRIPTION
**Pull request by zdz2101**
Per #60 and #64 , users seem to encounter this error with relative frequency: 

"Error: Writing failure: A provided format string could not be understood."  

I believe this is actually a `haven` error-message coming from line 55 in `R/write.R` :  
`write_xpt(data, path = path, version = 5, name = name)`

To remedy this, I propose to add an additional format check inside `xpt_validate()` which is called at line 47 of `R/write.R`, but changes are contained inside `R/utils-xportr.R`

### CODE REVIEW

**Comment by zdz2101** [comment]
@elimillera @bms63 Not quite sure how to diagnose the R-CMD-checks, besides that what do you think of these proposed changes?

**Comment by bms63** [comment]
Hey @zdz2101 I think I fixed the actions for linux os.  One is still running as I write

File: R/utils-xportr.R [hunk 1]
```diff
@@ -138,6 +161,12 @@ fmt_labs <- function(x) {
   glue("{labs} {encode_vals(val)}")
 }
 
+fmt_fmts <- function(x) {
+  fmts <- ntext(length(x), "Format", "Formats")
+  glue("{fmts} {encode_vals(x)}")
+}
+
+
```
Comment (bms63) [inline R/utils-xportr.R:169]: is this used somewhere?  I see it is new but don't see it being used?

**Review by bms63 — COMMENTED** [review]
Can you roxygen this file and make it so it doesn't export.  Sorry you have to clean up our mess!!

File: R/utils-xportr.R (hunk 1 above)
Comment (zdz2101) [reply R/utils-xportr.R:169]: Used up around line 114

Comment (bms63) [reply R/utils-xportr.R:169]: oh lol i see can you move it so it comes before that call

Comment (zdz2101) [reply R/utils-xportr.R:169]: If you scroll up a bit, you'll see a few other functions that are also mentioned after where they are called, do you want me to reorganize the entirety of the file such that functions are created **before** where they are called?

**Review by elimillera — APPROVED** [review]
Thanks @zdz2101! I'll be pushing up a some changes to the news file and description but otherwise I think this looks great.

File: R/utils-xportr.R (hunk 1 above)
Comment (bms63) [reply R/utils-xportr.R]: Please just to increase readability for those who come afterwards.

**Comment by bms63** [comment]
@zdz2101 Sorry.  The newly roxygen files shouldn't be exported unless you think they might be useful to users.  I just wanted titles, explanations of function purpose (if you think it is not clear to a new developer) and arguments described properly.  Like we do in admiral

File: DESCRIPTION [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Package: xportr
 Title: Utilities to Output CDISC SDTM/ADaM XPT Files
-Version: 0.1.0
+Version: 0.1.0.9000
```
Comment (bms63) [inline DESCRIPTION:3]: Can we just do 0.2.0 for our version?

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# xportr 0.1.0.9000 (development)
+* Added a new validation test that errors when users pass invalid formats. Thanks to @zdz2101!
```
Comment (bms63) [inline NEWS.md:2]: Can we link the issue here as well?

**Comment by zdz2101** [comment]
> @zdz2101 Sorry. The newly roxygened functions shouldn't be exported unless you think they might be useful to users. I just wanted titles, explanations of function purpose (if you think it is not clear to a new developer) and arguments described properly. Like we do in admiral

Ah I see I thought you meant it just isn't exported in the NAMESPACE, which it currently isn't. Although, I think by using roxygen-style comments and a command like `devtools::document()` it expects to create `.Rd` files and request appropriate entries in the reference section under the `pkgdown.yml` . I can leave the roxygen comments and use something like "@noRd" ?

**Comment by bms63** [comment]
> > @zdz2101 Sorry. The newly roxygened functions shouldn't be exported unless you think they might be useful to users. I just wanted titles, explanations of function purpose (if you think it is not clear to a new developer) and arguments described properly. Like we do in admiral
> 
> Ah I see I thought you meant it just isn't exported in the NAMESPACE, which it currently isn't. Although, I think by using roxygen-style comments and a command like `devtools::document()` it expects to create `.Rd` files and request appropriate entries in the reference section under the `pkgdown.yml` . I can leave the roxygen comments and use something like "@"noRd ?

Apologies for creating confusion.  Yes I meant `noRd` that is my bad.

File: NEWS.md (hunk 3 above)
Comment (zdz2101) [reply NEWS.md]: @elimillera

File: DESCRIPTION (hunk 2 above)
Comment (bms63) [reply DESCRIPTION:3]: oh  can you also add your name as a developer @zdz2101  ??

**Review by bms63 — APPROVED** [review]
I think this is looking great.  Thanks a lot @zdz2101!!  Just the version comment needs to be addressed by @elimillera

File: NEWS.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+# xportr 0.1.0.9000 (development)
```
Comment (zdz2101) [inline NEWS.md:1]:
```suggestion
# xportr 0.2.0
```

To match DESCRIPTION

**Pull request merged**